Bottom: 9cdfa8fc338b5927493ad5770aa1d2d9bd7f4cc6
Top:    886009990f4d52636c380f5b4c4f051be1a8608f
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-07-26 22:59:56 -0700

Refresh of mpi-finalize

---

diff --git a/src/rsocket.c b/src/rsocket.c
index d374af7..784d3f8 100644
--- a/src/rsocket.c
+++ b/src/rsocket.c
@@ -132,6 +132,7 @@ union rs_wr_id {
 	};
 };
 
+#define RS_RECV_WR_ID ~NULL
 /*
  * rsocket states are ordered as passive, connecting, connected, disconnected.
  */
@@ -449,7 +450,7 @@ static int rs_create_ep(struct rsocket *rs)
 		return ret;
 
 	for (i = 0; i < rs->rq_size; i++) {
-		ret = rdma_post_recvv(rs->cm_id, NULL, NULL, 0);
+		ret = rdma_post_recvv(rs->cm_id, RS_RECV_WR_ID, NULL, 0);
 		if (ret)
 			return ret;
 	}
@@ -881,7 +882,7 @@ static int rs_poll_cq(struct rsocket *rs)
 	int ret, rcnt = 0;
 
 	while ((ret = ibv_poll_cq(rs->cm_id->recv_cq, 1, &wc)) > 0) {
-		if (wc.opcode == IBV_WC_RECV_RDMA_WITH_IMM) {
+		if (wc.wr_id == RS_RECV_WR_ID) {
 			if (wc.status != IBV_WC_SUCCESS)
 				continue;
 			rcnt++;
@@ -923,7 +924,7 @@ static int rs_poll_cq(struct rsocket *rs)
 
 	if (rs->state & rs_connected) {
 		while (!ret && rcnt--)
-			ret = rdma_post_recvv(rs->cm_id, NULL, NULL, 0);
+			ret = rdma_post_recvv(rs->cm_id, RS_RECV_WR_ID, NULL, 0);
 
 		if (ret) {
 			rs->state = rs_error;
